ENIGMAsystem/SHELL/Bridges/General/Win32-Direct3D11.cpp

```diff
@@ -58,214 +55,60 @@ namespace enigma
-
-    // Set to a single back buffer.
+    DXGI_SWAP_CHAIN_DESC swapChainDesc = { };
     swapChainDesc.BufferCount = 1;
-
-    // Set the width and height of the back buffer.
     swapChainDesc.BufferDesc.Width = screenWidth;
     swapChainDesc.BufferDesc.Height = screenHeight;
-
-    // Set regular 32-bit surface for the back buffer.
     swapChainDesc.BufferDesc.Format = DXGI_FORMAT_R8G8B8A8_UNORM;
-
-    // Set the refresh rate of the back buffer.
-    if(m_vsync_enabled)
-    {
-      swapChainDesc.BufferDesc.RefreshRate.Numerator = numerator;
-      swapChainDesc.BufferDesc.RefreshRate.Denominator = denominator;
-    }
-    else
-    {
-      swapChainDesc.BufferDesc.RefreshRate.Numerator = 0;
-      swapChainDesc.BufferDesc.RefreshRate.Denominator = 1;
-    }
-
-    // Set the usage of the back buffer.
+    swapChainDesc.BufferDesc.RefreshRate.Numerator = 0;
+    swapChainDesc.BufferDesc.RefreshRate.Denominator = 1;
+    swapChainDesc.BufferDesc.ScanlineOrdering = DXGI_MODE_SCANLINE_ORDER_UNSPECIFIED;
+    swapChainDesc.BufferDesc.Scaling = DXGI_MODE_SCALING_UNSPECIFIED;
     swapChainDesc.BufferUsage = DXGI_USAGE_RENDER_TARGET_OUTPUT;
-
-    // Set the handle for the window to render to.
-    swapChainDesc.OutputWindow = hwnd;
-
-    // Turn multisampling off.
     swapChainDesc.SampleDesc.Count = 1;
     swapChainDesc.SampleDesc.Quality = 0;
-
-    // Set to full screen or windowed mode.
-    if(fullscreen)
-    {
-      swapChainDesc.Windowed = false;
-    }
-    else
-    {
-      swapChainDesc.Windowed = true;
-    }
-
-    // Set the scan line ordering and scaling to unspecified.
-    swapChainDesc.BufferDesc.ScanlineOrdering = DXGI_MODE_SCANLINE_ORDER_UNSPECIFIED;
-    swapChainDesc.BufferDesc.Scaling = DXGI_MODE_SCALING_UNSPECIFIED;
-
-    // Discard the back buffer contents after presenting.
+    swapChainDesc.OutputWindow = enigma::hWnd;
+    swapChainDesc.Windowed = true; // initially windowed and not fullscreen
     swapChainDesc.SwapEffect = DXGI_SWAP_EFFECT_DISCARD;
-
-    // Don't set the advanced flags.
     swapChainDesc.Flags = 0;
 
-    // Set the feature level to DirectX 11.
-    featureLevel = D3D_FEATURE_LEVEL_11_0;
-
-    // Create the swap chain, Direct3D device, and Direct3D device context.
+    D3D_FEATURE_LEVEL featureLevel = D3D_FEATURE_LEVEL_11_0;
+    #ifdef DEBUG_MODE
+    result = D3D11CreateDeviceAndSwapChain(NULL, D3D_DRIVER_TYPE_HARDWARE, NULL, D3D11_CREATE_DEVICE_DEBUG, &featureLevel, 1,
+                                           D3D11_SDK_VERSION, &swapChainDesc, &m_swapChain, &m_device, NULL, &m_deviceContext);
+    #else
     result = D3D11CreateDeviceAndSwapChain(NULL, D3D_DRIVER_TYPE_HARDWARE, NULL, 0, &featureLevel, 1,
-                   D3D11_SDK_VERSION, &swapChainDesc, &m_swapChain, &m_device, NULL, &m_deviceContext);
+                                           D3D11_SDK_VERSION, &swapChainDesc, &m_swapChain, &m_device, NULL, &m_deviceContext);
+    #endif
```

Comment:
@JoshDreamland is this macro the cleanest way for me to specify `D3D11_CREATE_DEVICE_DEBUG` or could this be a few lines less?

Reply:
Since we're just checking whether DEBUG_MODE is defined, not whether it's true, this is approximately the best way. You could create a `constexpr bool kDebugMode` up top, and then pass `kDebugMode ? D3D11_CREATE_DEVICE_DEBUG : 0`.

Reply:
Alright, I decided to use constexpr for this: 8a204a2870b1f9a2dc40b3c53f3c365f318bc5e2

Thanks, the other way was just too ugly for my taste. This only occurs once too anyway so I mean, checking the extra variable isn't a big deal.

Reply:
Actually, I changed my mind and decided to just make `flags` a stack-allocated variable: 16c1d801f7eda4fb9de10b9f406633e14f555d7d

I've done it this way in other places, like for the vertex shader, and it's my preferred method of doing it.